pkg/didcomm/transport/http/inbound.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package http
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+// inboundCommHTTPOpts struct for path and handler represented
+// as an inbound http option
+type inboundCommHTTPOpts struct {
+	Path        string
+	HandlerFunc http.HandlerFunc
+}
+
+// InboundHTTPSetting is an inbound HTTP transport options setting function
+type InboundHTTPSetting func(opts *inboundCommHTTPOpts)
+
+// InboundDIDCommHandler struct to pass details for request handlers
+type InboundDIDCommHandler struct {
+	handlerOpts *inboundCommHTTPOpts
+}
+
+// WithInboundSetting option is for creating an Inbounnd HTTP transport using an http.Handler
+func WithInboundSetting(inboundPath string, handlerFunc http.HandlerFunc) InboundHTTPSetting {
+	return func(opts *inboundCommHTTPOpts) {
+		opts.Path = inboundPath
+		opts.HandlerFunc = handlerFunc
+	}
+}
+
+// NewInboundHandler will create a new handler to enforce Did-Comm HTTP transport specs
+// then routes processing to the passed 'in' handler argument
+// 'in' is mandatory as it contains the url Path and the HTTP handler needed by inbound comm.
+func NewInboundHandler(in InboundHTTPSetting) (http.Handler, error) {
```

Comment:
The purpose of an Options struct/method pattern is to allow 0 or more option methods. Why does the constructor only take one? Is this argument optional or required?

Reply:
removed the struct altogether, now the function accepts only a url and an http handlerFunc

Reply:
now the function accepts a msgHandler function